### ISSUE TITLE
10X genomics BAM with tags not working with dropest

### ISSUE THREAD

**Issue by Jemkon** [hms-dbmi/dropEst#18]
Hello,

I have a BAM from cellranger 10x tool. I would like to use it for velocyto and therefore want to create input file using dropest. 

Here is the command I use:
`dropest -f -c 10x.xml 10x_001/001_possorted_genome_bam.bam`
it runs successfully and created cell.counts.matrices.rds and cell.counts.rds files. The matrix only contains Exonic reads info and when I load the cell.counts.matrices.rds file for velocity it doesnt work as there is no data for intronic and intergenic. 

Although I looked my BAM and found the reads for intronic and intergenic regions.

Any idea why dropest only selecting exonic reads?

Many thanks.

**Comment by VPetukhov** [comment]
Hello,

Please check that your tags for intronic and exonic reads are the same as in the pipeline (see "Estimation/BamTags/Type" section of config_desc.xml)

**Comment by Jemkon** [comment]
Hi,

Many thanks for replying to my question. The BAM tag is RE:A:E (Exonic), RE:A:N (intronic) and RE:A:I (intergenic).

Here is the BAM :
`M01145:3:000000000-BLL58:1:1106:13817:2692      256     1       14145   0       50M     *       0
       0       ATTAGGCTCTCAGCATGACTATTTTTAGAGACCCCGTGTCTGTCACTGAA      CCCCBFFFFFFFFGGGGGGGGGHHHHHHHHHHGHGGGGHHHHFHHHHHHF      NH:i:5  HI:i:2  AS:i:49 nM:i:0  RG:Z:id_11189GApool07__TVsc10x_007:MissingLibrary:1:000000000-BLL58:1   RE:A:I  BC:Z:AGCCAATT   QT:Z:DDCDDFFF   CR:Z:ACTGATGTCTGGAGCC   CY:Z:CDDDDFFFFFFFGGGG   CB:Z:ACTGATGTCTGGAGCC-1 UR:Z:GGAGTAACTG UY:Z:GGGGEGHHHH UB:Z:GGAGTAACTG
M01145:3:000000000-BLL58:1:1111:8218:23302      272     1       14404   0       50M     *       0
       0       TTTCTGCTCAGTTCTTTATTGATTGGTGTGCCGTTTTCTCTGGAAGCCTC      FHHHHHGGHHGHHGHHHGHHHFFHGHHGGGCGGGGGGGFFFFFFBABBBB      NH:i:8  HI:i:3  AS:i:49 nM:i:0  RG:Z:id_11189GApool07__TVsc10x_007:MissingLibrary:1:000000000-BLL58:1   RE:A:I  BC:Z:AGCCAATT   QT:Z:BCCABFFF   CR:Z:ACAGCCGAGCGGATCA   CY:Z:BCCCCDDDDCCCGGGG   CB:Z:ACAGCCGAGCGGATCA-1 UR:Z:GACAGCCTGG UY:Z:GGGGGGHHHH UB:Z:GACAGCCTGG
M01145:3:000000000-BLL58:1:1110:28307:13643     272     1       14404   0       3S47M   *       0
       0       TTTTTTTTGCTCATTTCTTTATTGATTGGTGTGCCGTTTTCTCTGGAAGC      GGGHGB0B11D110GF1HHF1HGCCHGAGGEEA11GGGFFAFFFC>11A>      NH:i:9  HI:i:3  AS:i:42 nM:i:2  RG:Z:id_11189GApool07__TVsc10x_007:MissingLibrary:1:000000000-BLL58:1   RE:A:I  BC:Z:CTAGGTGA   QT:Z:DCDDDFFF   CR:Z:TCGTACCTCAAACCGT   CY:Z:CCCBBBCFFFFFGGGG   CB:Z:TCGTACCTCAAACCGT-1 UR:Z:TGGCGAGAGT UY:Z:GGGGGGGGGH UB:Z:TGGCGAGAGT
`
Here is the config.xml
`<BamTags> 
            <cb>CB</cb> 
            <umi>UB</umi> 
            <gene>GX</gene> 
			<cb_quality>CQ</cb_quality> 
            <umi_quality>UQ</umi_quality> 
            <Type> 
                <tag>RE</tag>
                <intronic>N</intronic> 
                <intergenic>I</intergenic> 
                
            </Type>
        </BamTags>`

Am I doing anything wrong here?

Thanks.

**Comment by VPetukhov** [comment]
Wou, can you please paste it in more readable form?

**Comment by Jemkon** [comment]
Sorry for the mess. I have updated the blocks.
thanks.

**Comment by VPetukhov** [comment]
Hm, looks like it should work. Though, your bam has tag type "A", which is used for single characters, but I use std::string in the pipeline. Not sure if BamTools processes it correctly. I'll try to check it this week. Now you can just use custom gene annotation through "-g" option.

**Comment by Jemkon** [comment]
I have tried with option "-g" but got errors. Although i have used the same gtf as I used it for cellranger.
Command:
```
dropest -V -m -g /hg38/cellRanger/refdata-cellranger-GRCh38-1.2.0/genes/genes.gtf -c /configs/10x.xml /id_pool07__10x_007/outs/possorted_genome_bam.bam

```

Errors:
```
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:2108:16038:21056
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:1102:25532:11813
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:1105:20253:10236
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:2113:24507:19610
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:2108:26090:15148
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:2110:28447:17362
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:1108:13628:8409
ERROR: unable to parse out UMI in: M01145:3:000000000-BLL58:1:2105:3427:9825
ERROR: can't find chromosome, id: -1
/id_pool07__10x_007/outs/possorted_genome_bam.bam: 5723859 total reads; 0% intergenic; 0% touch exon; 0% touch intron; 0% touch not annotated regions; can't parse 55.7132% reads; 0 CBs read
Bams parsed: 17:00:55.

Compiling intron/exon matrices: 17:00:55.
Exon
0 genes, 0 cells.
Intron
0 genes, 0 cells.
Intron/exon spanning
0 genes, 0 cells.
Done: 17:00:55.

```

**Comment by VPetukhov** [comment]
You still need to use "-f" argument.

**Comment by Jemkon** [comment]
Many thanks. indeed It works...
Could you please let me know once you fix the BAM tag issue?

**Comment by VPetukhov** [comment]
Yes, sure, I'll write in this thread.

**Comment by VPetukhov** [comment]
It was BamTools' problem. Fix is now in the develop branch (14a10a8f8f030626dd1672d5d81fce0cbdd971f9), I plan to merge it to the master this week.
